srml/staking/src/lib.rs

```diff
@@ -1380,3 +1380,17 @@ impl<T: Trait> SelectInitialValidators<T::AccountId> for Module<T> {
 		<Module<T>>::select_validators().1
 	}
 }
+
+
+/// StakingSlasher
+pub struct StakingSlasher<T>(rstd::marker::PhantomData<T>);
+
+impl<T: Trait> DoSlash<Perbill, (T::AccountId, Exposure<T::AccountId, BalanceOf<T>>)> for StakingSlasher<T> {
+	fn do_slash(severity: Perbill, (who, exposure): (T::AccountId, Exposure<T::AccountId, BalanceOf<T>>)) {
```

Comment:
note, perbill is a concrete type here

Reply:
we also will want to reward those who reported the slash using some of the funds from the slash.

the simplest but probably least-flexible way is to have `do_slash` take another argument `who: Vec<(AccountId, Perbill)>` where the Perbill is a "priority". For a single reporter, this would be 100%. If we have many reporters (e.g. fishermen), we would distribute the reward among them unevenly based on who was first.